ql/patterns/singleton.hpp

```diff
@@ -26,6 +26,8 @@
 
 #include <ql/types.hpp>
 #include <boost/shared_ptr.hpp>
+#include <boost/atomic.hpp>
```

Comment:
@sdgit : can we include these two headers only if the thread-safe singleton is enabled?